rclcpp/include/rclcpp/node_interfaces/node_base_interface.hpp

```diff
@@ -142,9 +143,20 @@ class NodeBaseInterface
    */
   RCLCPP_PUBLIC
   virtual
-  rcl_guard_condition_t *
+  const rcl_guard_condition_t *
```

Comment:
Same here, it shouldn't be necessary if you can return a non-const pointer to the rclcpp guard condition.